### PR TITLE
Added NaturalInt and PositiveInt. fixes #27

### LINKED ISSUE #27: Add PositiveInt/NaturalInt

### PR DESCRIPTION
**Pull request by etorreborre**

### CODE REVIEW

File: disorder-core/src/Disorder/Core/NaturalInt.hs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import           Test.QuickCheck
+
+
+data NaturalInt = NaturalInt {
+    value :: Int
+  } deriving (Eq, Show)
+
+instance Arbitrary NaturalInt where
+  arbitrary = NaturalInt <$> (suchThat arbitrary ((<=) 0))
```
Comment (domdere) [inline disorder-core/src/Disorder/Core/NaturalInt.hs]: If you defined an `Integral` and `Bounded` instance for these you could just use `arbitrary = arbitrarySized Bounded Integral`, [doc here] (https://hackage.haskell.org/package/QuickCheck-2.4.0.1/docs/Test-QuickCheck.html#v:arbitrarySizedBoundedIntegral)

`suchThat` effectively does a filter, while the above, or `choose` will only select from valid elements

Comment (domdere) [inline disorder-core/src/Disorder/Core/NaturalInt.hs]: Actually an `Integral` instance is kind of lying with these types, but i think `choose` is still better with the upper bound being given by the `Int` `Bounded` instance

Comment (domdere) [inline disorder-core/src/Disorder/Core/NaturalInt.hs]: i.e

``` Haskell
instance Arbitrary NaturalInt where
    arbitrary =  NaturalInt <$> choose 0 maxBound
```

Comment (amosr) [inline disorder-core/src/Disorder/Core/NaturalInt.hs]: I wonder whether it is worth taking the current size into account?

```
instance Arbitrary NaturalInt where
    arbitrary =  sized $ \s -> NaturalInt <$> choose 0 s
```

**Comment by domdere** [comment]
:+1:

**Pull request merged**